plugin/src/main/groovy/com/btkelly/gnag/reporters/ViolationDetectorFactory.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2016 Bryan Kelly
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.btkelly.gnag.reporters;
+
+import com.btkelly.gnag.extensions.GnagPluginExtension;
+import com.btkelly.gnag.utils.ProjectHelper;
+
+import org.gradle.api.Project;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ViolationDetectorFactory {
+
+    public static List<ViolationDetector> getAllViolationDetector(Project project, GnagPluginExtension gnagPluginExtension) {
+        List<ViolationDetector> violationDetectors = new ArrayList<>();
+        violationDetectors.add(getCheckStyleViolationDetector(project, gnagPluginExtension));
+        violationDetectors.add(getPmdViolationDetector(project, gnagPluginExtension));
+        violationDetectors.add(getFindBugViolationDetector(project, gnagPluginExtension));
+        violationDetectors.add(getKtLintViolationDetector(project, gnagPluginExtension));
+        violationDetectors.add(getDetektViolationDetector(project, gnagPluginExtension));
+        violationDetectors.add(getAndroidLintViolationDetector(project, gnagPluginExtension));
+
+        return violationDetectors;
+    }
+
+    public static ViolationDetector getCheckStyleViolationDetector(Project project, GnagPluginExtension gnagPluginExtension) {
+        ProjectHelper projectHelper = new ProjectHelper(project);
+        boolean isEnabled = gnagPluginExtension.checkstyle.isEnabled() && projectHelper.hasJavaSourceFiles();
+        return new CheckstyleViolationDetector(project, isEnabled, gnagPluginExtension.checkstyle);
```

Comment:
I feel like if a detector is disabled the task should not show up in the list of available tasks as executing it would do nothing.

Reply:
You are right.